### PR TITLE
docs: Update info on JDTLS install for Java

### PR DESCRIPTION
**Pull request by valentinegb**
@notpeter @ABckh @HeavySnowJakarta really sorry to open yet another PR for the Java docs, but the Zed Java extension now auto-downloads JDTLS (surprise!) so the information related to that in the docs is no longer accurate.

Release Notes:

- N/A

### CODE REVIEW

File: docs/src/languages/java.md [hunk 1]
```diff
@@ -88,15 +103,16 @@ If you have issues with either of these plugins, please open issues on their res
 
 ## Example Configs
 
-### Zed Java Classpath
+### Zed Java Settings
 
-You can optionally configure the class path that JDTLS uses with:
+You can optionally configure the version of JDTLS to download (see <#optional-install-jdtls>) or the class path that JDTLS uses with:
```
Comment (valentinegb) [inline docs/src/languages/java.md:108]: The `<#optional-install-jdtls>` part probably needs to fix, but I'm not sure how. Hopefully you get the idea here.

**Comment by ABckh** [comment]
> These are similar in that this PR uses the Eclipse JDT Language Server which zed-java-eclipse-jdtls also uses. They are different in I'm sure many small ways (simply differing implementations of completion labels for instance) but most notably @ABckh maintains a repository to allow for zed-java-eclipse-jdtls to download JDTLS for you, while this PR would have zed-extensions/java ask you to install JDTLS yourself (it's available on Homebrew). Also, this PR adds a setting to let you change the Java binary that JDTLS will use (for different versions of Java, for example).

Hi @valentinegb! What is the difference between the extensions now then?
I am just really curious that we do the same stuff twice, the history was:
1) There was `zed-java-eclipse-jdtls` which used jdtls
2) Then you added jdtls to `zed -java` and pointed that the main difference to keep both is that one of them do not install jdtls automatically
3) You added auto installation of jdtls to `zed-java`

Now after you have added the same stuff to `zed-java` it becomes the same as `zed-java-eclipse-jdtls`. I am just a bit disappointed with how things work done with java extensions

**Comment by notpeter** [comment]
Thanks for this.  I moved the section on manually installing to the bottom and consolidated the examples a bit.  Feel free to keep additions/changes coming.  We've adjusted the CI so docs-only PRs only take 15seconds to re-test so it's easier to make small incremental improvements.

**Comment by valentinegb** [comment]
@ABckh, I do apologize that there's now only really nuanced differences between the extensions, but the nuances do exist, and I hope that you can consider this just some friendly competition ^^

And @notpeter, thanks for your help through all this!

**Comment by ABckh** [comment]
> @ABckh, I do apologize that there's now only really nuanced differences between the extensions, but the nuances do exist, and I hope that you can consider this just some friendly competition ^^
> 
> And @notpeter, thanks for your help through all this!

No need to apologize. This is why I’m disappointed—it doesn’t feel like fair competition. zed-java gained popularity early on, despite not working well, likely due to historical reasons. Then we built something that works, and it started gaining traction (8k downloads so far). However, all of our features were copied into the original extension. People will likely install the original extension since it has more downloads and now the same features. They don’t care about competition, and zed-java-eclipse-jdtls won’t get the attention it deserves.

The real issue is that I’ve lost motivation to work on new features. If I add something new, it will just get copied into the zed-java extension anyway. Maybe this isn’t just a problem with how Zed manages extensions, but a broader issue with open source in general.  🙂

**Comment by valentinegb** [comment]
@ABckh I get what you mean... I honestly would really like if we would both work on zed-java instead of maintaining separate extensions, that way, with your help, we can have a Java extension that's as good as can be, while also making sure all the features are taken advantage of by all Zed Java devs. That was my goal when I started contributing to zed-java, and I mostly achieved that, but it's maybe not "as good as can be" yet and there's still a split between what extension Java devs use, although both extensions are pretty good, so the split isn't quite as important now. You expressed before that you wanted to keep working on zed-java-eclipse-jdtls, so I had decided that I wasn't going to try to convince you to work on zed-java instead.

So, it is a bit unfair, and that's not how I meant for it to be, but that is how it ended up. I don't make decisions for you of course, but I do think it's in both of our best interests if you worked on zed-java instead of zed-java-eclipse-jdtls, although I will do my best to respect whatever decision you make.

**Pull request merged**